RIOT-OS/RIOT
# dist/tools/buildsystem_sanity_check: check no PORT exports

### Contribution description

Add sanity check for removed exports.

PORT is not exported anymore since https://github.com/RIOT-OS/RIOT/pull/10440 so prevent for re-appearing.

### Testing procedure

```
./dist/tools/buildsystem_sanity_check/check.sh
```
Returns no errors. (It is executed by the static tests too so would detect issues).

If adding this diff:

``` diff
diff --git a/Makefile b/Makefile
index 144cbd7e6..fe51b1bda 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,9 @@
 .all:
+# Nothing after: export PORT
+export PORT = space
+export PORT?= question_mark
+export PORT= equal
+export PORT:= colon
 
 .PHONY: all doc doc-man doc-latex docclean print-versions welcome

```

All the lines are detected

```
./dist/tools/buildsystem_sanity_check/check.sh 
Invalid build system patterns found by ./dist/tools/buildsystem_sanity_check/check.sh:
Variables must not be exported:
	Makefile:# Nothing after: export PORT
	Makefile:export PORT = space
	Makefile:export PORT?= question_mark
	Makefile:export PORT= equal
	Makefile:export PORT:= colon
```

### Issues/PRs references

Follow-up to https://github.com/RIOT-OS/RIOT/pull/10440 and part of https://github.com/RIOT-OS/RIOT/issues/10850
